clang/lib/Sema/SemaChecking.cpp

```diff
@@ -13815,15 +13815,32 @@ static void CheckImplicitConversion(Sema &S, Expr *E, QualType T,
         Expr::EvalResult result;
         if (E->EvaluateAsRValue(result, S.Context)) {
           // Value might be a float, a float vector, or a float complex.
-          if (IsSameFloatAfterCast(result.Val,
-                   S.Context.getFloatTypeSemantics(QualType(TargetBT, 0)),
-                   S.Context.getFloatTypeSemantics(QualType(SourceBT, 0))))
+          if (IsSameFloatAfterCast(
+                  result.Val,
+                  S.Context.getFloatTypeSemantics(QualType(TargetBT, 0)),
+                  S.Context.getFloatTypeSemantics(QualType(SourceBT, 0)))) {
+            if (S.getLangOpts().SYCLIsDevice)
+              S.SYCLDiagIfDeviceCode(CC, diag::warn_imp_float_size_conversion);
+            else
+              DiagnoseImpCast(S, E, T, CC,
+                              diag::warn_imp_float_size_conversion);
             return;
+          }
         }
 
         if (S.SourceMgr.isInSystemMacro(CC))
           return;
-
+        // If there is a precision conversion between floating point types when
+        // -Wimplicit-float-size-conversion is passed but
+        // -Wimplicit-float-conversion is not, make sure we emit atleast a size
+        // warning.
+        if (S.Diags.isIgnored(diag::warn_impcast_float_precision, CC) &&
+            !S.Diags.isIgnored(diag::warn_imp_float_size_conversion, CC)) {
```

Comment:
```suggestion
        if (S.Diags.isIgnored(diag::warn_impcast_float_precision, CC)) {
```
I am pretty sure this is equivalent -- if the second diagnostic is ignored, the diagnostics engine handles the suppression of the diagnostic, so I don't think you need to check it here as well (there's no real performance benefits to doing so).